### DISCUSSION TITLE
usePrepareContractWrite() populated with useContractRead() fails

### DISCUSSION THREAD

**Discussion by Fabien-Morrow** [wevm/wagmi#1149] [Q&A]
I don't understand why this snippet doesn't work :/
My _app.js is quite standart, with rainbowkit on goerli and alchemy provider with my key

You can launch the code below here :
[https://codesandbox.io/p/sandbox/usepreparecontractwrite-mwe-y92p19](https://codesandbox.io/p/sandbox/usepreparecontractwrite-mwe-y92p19)

step to reproduce :
- connect your wallet
- an error throws because of "value: ethers.utils.parseUnits(data)". data is retrieved by useContractRead() but is undefined, I don't know why !
- change value to another one for example "value : "30",
- this time the page loads and useContractRead() retrieve data !
- if you change value one more time to "value : ethers...." it works....

Why the code doesn't work on the first load ?

```js
import React from "react";
import { ethers } from "ethers";
import { ConnectButton } from "@rainbow-me/rainbowkit";
import { useAccount, useContractRead, usePrepareContractWrite } from "wagmi";

const abi = [
  {
    inputs: [],
    name: "price",
    outputs: [
      {
        internalType: "uint256",
        name: "",
        type: "uint256",
      },
    ],
    stateMutability: "view",
    type: "function",
  },
  {
    inputs: [
      {
        internalType: "uint256",
        name: "amount",
        type: "uint256",
      },
    ],
    name: "mint",
    outputs: [],
    stateMutability: "payable",
    type: "function",
  },
];

function Mint() {
  const { data, isSuccess, isLoading } = useContractRead({
    addressOrName: "0x575f791cF0e9Ca90f09BC150F03817c453B5b41C",
    contractInterface: abi,
    functionName: "price",
  });
  console.log(data, isSuccess, isLoading);
  const { config } = usePrepareContractWrite({
    addressOrName: "0x575f791cF0e9Ca90f09BC150F03817c453B5b41C",
    contractInterface: abi,
    functionName: "mint",
    args: [1],
    overrides: {
      // 0 - Initial config : value : ethers... it fails !
      // 1 - replace value with 30 and it works.
      // 2 - replace 30 with ethers.utils... and now it works !

      // value: "30",
      value: ethers.utils.parseUnits(data),
    },
  });
  return <div>mint</div>;
}

export default function Home() {
  const [hasMounted, setHasMounted] = React.useState(false);
  const { isConnected } = useAccount();

  React.useEffect(() => setHasMounted(true), []);
  return (
    <div>
      <ConnectButton />
      {hasMounted && isConnected && <Mint />}
    </div>
  );
}

```

**Comment by RogerPodacter** [comment]
What if you add `enabled: Boolean(data)` to the `usePrepareContractWrite`?

**Reply by Fabien-Morrow** [reply]
Thanks for your support but it doesn't work :/
I don't know if it's a bad usage of the lib from my side, or if this can be considered as a bug... I'll open an issue mid week if I'm still stuck on this.

**Comment by Fabien-Morrow** [comment]
You were right man...
I needed `enabled: Boolean(data)` but I forgot price is already a BigNumber in my mwe, so the `parseUnits ` part was not needed !

with :
```js
  const { config } = usePrepareContractWrite({
    addressOrName: "0x575f791cF0e9Ca90f09BC150F03817c453B5b41C",
    contractInterface: abi,
    functionName: "mint",
    enabled: Boolean(data),
    args: [1],
    overrides: {
      value: data ? data : undefined,
    },
  });
```
It works !!

**Reply by RogerPodacter** [reply]
Nice! I think you can even just do `value: data` as in the case that data doesn't exist the hook will be disabled and won't try to evaluate the overrides. Your approach is quite explicit though which is good! Glad you found the answer.
